PopPUNK/plot.py

```diff
@@ -728,12 +721,64 @@ def outputsForMicroreact(combined_list, coreMat, accMat, clustering, perplexity,
     writeClusterCsv(outPrefix + "/" + os.path.basename(outPrefix) + "_microreact_clusters.csv",
                         combined_list, combined_list, clustering, 'microreact', epiCsv, queryList)
 
+    # write NJ tree
+    if nj_tree is not None:
+        write_tree(nj_tree, outPrefix, "_core_NJ.nwk", overwrite)
+
+    # write MST
+    if mst_tree is not None:
+        write_tree(mst_tree, outPrefix, "_core_MST.nwk", overwrite)
 
     # write the phylogeny .nwk; t-SNE network .dot; clusters + data .csv
-    generate_phylogeny(coreMat, seqLabels, outPrefix, "_core_NJ.nwk", rapidnj, overwrite, threads)
     generate_tsne(seqLabels, accMat, perplexity, outPrefix, overwrite)
 
-def generate_phylogeny(coreMat, seqLabels, outPrefix, tree_suffix, rapidnj, overwrite, threads):
+def write_tree(tree, prefix, suffix, overwrite):
+    """Generate phylogeny using dendropy or RapidNJ
+
+    Writes a neighbour joining tree (.nwk) from core distances.
+
+    Args:
+        tree (str)
+            Newick-formatted string representation of tree
+        prefix (str)
+            Prefix for output files
+        suffix (str)
+            Suffix for output files
+        overwrite (bool)
+            Whether to overwrite existing files
+    """
+    fn = prefix + '/' + prefix
+    tree_filename = prefix + "/" + os.path.basename(prefix) + suffix
+    if overwrite or not os.path.isfile(tree_filename):
+        with open(tree_filename,'w') as tree_file:
+            tree_file.write(tree)
+
+def check_tree_exists(prefix, type):
+    """Checks for existing trees from previous runs.
+
+    Args:
+        prefix (str)
+            Output prefix used for search
+        type (str)
+            Type of tree (NJ or MST)
+            
+    Returns:
+        tree_string (str)
+            Newick-formatted string of NJ tree
+    """
+    tree_string = None
+    tree_prefix = prefix + "/" + os.path.basename(prefix)
+    for suffix in ["_core_" + type + ".tree","_core_" + type + ".nwk"]:
+            tree_fn = tree_prefix + suffix
```

Comment:
Does this need a `"/"` in the middle? Consider `os.path.join()`

Reply:
Changed in ce72382.